### PR TITLE
Sdem/backward coupling tests

### PR DESCRIPTION
**Pull request by GuillermoCasas**
**Description**
Adding a first backward-coupling test, which consists in making sure the solid fraction is conservative in the sense that the integral of the solid volume fraction recovers the volume of the spheres present. 

**Changelog**
Basically adding the infrastructure in analogy to preexisting tests. The idea is to have further tests added in this category soon. In particular, some testing the time-filtered version of the backward-coupling.

### CODE REVIEW

**Comment by maceligueta** [comment]
For some reason on test is failing in Windows:
 ```
File "D:\a\Kratos\Kratos\bin\Custom\applications\SwimmingDEMApplication\tests\SmallTests.py", line 87, in <module>
    available_tests += [test_class for test_class in CandelierTF.TestFactory.__subclasses__()]
NameError: name 'CandelierTF' is not defined
```

File: applications/SwimmingDEMApplication/tests/SmallTests.py [hunk 1]
```diff
@@ -25,8 +25,15 @@
      fluid_DEM_coupling_imports_available = True
 except ImportError:
      fluid_DEM_coupling_imports_available = False
-# List of tests that are available
-available_tests = []
+
+from KratosMultiphysics import Vector, Logger, Parameters
+
+
+def Say(*args):
```
Comment (jgonzalezusua) [inline applications/SwimmingDEMApplication/tests/SmallTests.py]: Is this function used somewhere?

File: applications/SwimmingDEMApplication/tests/SmallTests.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Definition of the classes for the SMALL TESTS
 
-#Iimport Kratos
+# Import Kratos
```
Comment (jgonzalezusua) [inline applications/SwimmingDEMApplication/tests/SmallTests.py]: If this import is commented you can remove it

**Comment by jgonzalezusua** [comment]
@maceligueta , this due to he moved addition of the available_tests out of the conditions in the _SmallTests.py_

File: applications/SwimmingDEMApplication/tests/SmallTests.py [hunk 3]
```diff
@@ -67,14 +75,17 @@ class candelier_with_history_hinsberg_test(CandelierTF.TestFactory):
...
 if fluid_DEM_coupling_imports_available:
      class fluid_dem_coupling_one_way_test(FDEMTF.TestFactory):
           file_name = "fluid_dem_tests/settling_cube"
           file_parameters = "fluid_dem_tests/ProjectParameters.json"
 
-     available_tests += [test_class for test_class in FDEMTF.TestFactory.__subclasses__()]
+available_tests = []
+available_tests += [test_class for test_class in InterpolationTF.TestFactory.__subclasses__()]
+available_tests += [test_class for test_class in BackwardCouplingTF.TestFactory.__subclasses__()]
+available_tests += [test_class for test_class in CandelierTF.TestFactory.__subclasses__()]
```
Comment (jgonzalezusua) [inline applications/SwimmingDEMApplication/tests/SmallTests.py]: I think these lines have to be inside the conditions you set to define the subclasses. In fact, you are not able to import the CandelierTF, so your code goes to line 22 and it does not define the classes of the candelier tests. However, you are appending them in that line, so the code crashes. You should undo this approach.

**Comment by jgonzalezusua** [comment]
@GuillermoCasas your branch has conflicts, you should resolved them if you want CI to run

File: applications/SwimmingDEMApplication/tests/SmallTests.py (hunk 3 above)
Comment (GuillermoCasas) [reply applications/SwimmingDEMApplication/tests/SmallTests.py]: But if the import was not possible, then there are no subclasses appended. I do not see the problem.

Comment (GuillermoCasas) [reply applications/SwimmingDEMApplication/tests/SmallTests.py]: Oh, yes. You are right.

**Comment by jgonzalezusua** [comment]
You can address the Codacy warnings but it is not compulsory. As far as I am concerned, this PR can be approved.

**Review by jgonzalezusua — DISMISSED** [review]

**Review by jgonzalezusua — DISMISSED** [review]

**Review by jgonzalezusua — APPROVED** [review]

**Pull request merged**